### PR TITLE
IOS-0952 :: MJ Harmony :: Feature Internal :: Implement generic network data source

### PR DESCRIPTION
**Pull request by newboadki**
* Replace Alamofire by URLSession in Harmony.
* Asana task link: https://app.asana.com/0/1109863238977521/1202971884260952/f

### CODE REVIEW

**Review by joselufo — APPROVED** [review]

File: Example/Harmony/Core/Features/Item/ItemProvider.swift [hunk 1]
```diff
@@ -17,14 +16,14 @@ class ItemDefaultModule: ItemComponent {
...
-    init(executor: Executor, apiClient: Session, storage: AnyDataSource<Data>) {
+    init(executor: Executor, apiClient: URLSession, storage: AnyDataSource<Data>) {
         self.executor = executor
         self.storage = storage
     }
     
     private lazy var networkDataSource: AnyDataSource<ItemEntity> = {
-        let baseDataSource = GetNetworkDataSource<ItemEntity>(url: "https://demo3068405.mockable.io/", session: Session.default, decoder: JSONDecoder())
-        
+        let baseDataSource = GetNetworkDataSource<ItemEntity>(url: URL(string:"https://demo3068405.mockable.io/")!, session: URLSession.shared, decoder: JSONDecoder())
```
Comment (vilanovi) [inline Example/Harmony/Core/Features/Item/ItemProvider.swift]: You're doing a reference here to a `URLSession.shared` object, but you have in ApplicatoinDefaultModule, an `apiClient: URLSession` object that you instantiate there. Shouldn't it be that one instead?

**Review by vilanovi — APPROVED** [review]
Added one comment, but approved the PR. Up to you to decide what to do with my comment (if you want to change it or not).

File: Example/Harmony/Core/Features/Item/ItemProvider.swift (hunk 1 above)
Comment (newboadki) [reply Example/Harmony/Core/Features/Item/ItemProvider.swift]: Yes, done.

**Comment by joselufo** [comment]
@newboadki, everybody has approved the PR. Feel free to merge it.

**Pull request merged**